appliances/MinIO/appliance.sh

```diff
@@ -0,0 +1,396 @@
+ExecStartPre=/bin/bash -c "if [ -z \"${MINIO_VOLUMES}\" ]; then echo \"Variable MINIO_VOLUMES not set in /etc/default/minio\"; exit 1; fi"
+ExecStart=/usr/local/bin/minio server $MINIO_OPTS $MINIO_VOLUMES
+
+# MinIO RELEASE.2023-05-04T21-44-30Z adds support for Type=notify (https://www.freedesktop.org/software/systemd/man/systemd.service.html#Type=)
+# This may improve systemctl setups where other services use `After=minio.server`
+# Uncomment the line to enable the functionality
+# Type=notify
+
+# Let systemd restart this service always
+Restart=always
+
+# Specifies the maximum file descriptor number that can be opened by this process
+LimitNOFILE=65536
+
+# Specifies the maximum number of threads this process can create
+TasksMax=infinity
+
+# Disable timeout logic and wait until process is stopped
+TimeoutStopSec=infinity
+SendSIGKILL=no
+
+[Install]
+WantedBy=multi-user.target
+
+# Built for ${project.name}-${project.version} (${project.name})
+
+EOF
+
+    msg info "Enable systemd service for MinIO"
+    systemctl enable minio.service
+}
+
+create_default_environment_variable_file()
+{
+    msg info "Create default Environment Variable File for MinIO"
+    cat > /etc/default/minio <<'EOF'
+#
+# ATTENTION: This file is managed by MinIO Service Contextualization. DO NOT manually edit this file, instead
+# use contextualization parameters to update MinIO attributes.
+#
+# Begin Environment Variable File
+#
+# MINIO_ROOT_USER and MINIO_ROOT_PASSWORD sets the root account for the MinIO server.
+# This user has unrestricted permissions to perform S3 and administrative API operations on any resource in the deployment.
+# Omit to use the default values 'minioadmin:minioadmin'.
+# MinIO recommends setting non-default values as a best practice, regardless of environment
+
+MINIO_ROOT_USER=myminioadmin
+MINIO_ROOT_PASSWORD=minio-secret-key-change-me
+
+# MINIO_VOLUMES sets the storage volume or path to use for the MinIO server.
+
+MINIO_VOLUMES="/mnt/data"
+
+# MINIO_OPTS sets any additional commandline options to pass to the MinIO server.
+# For example, `--console-address :9001` sets the MinIO Console listen port
+MINIO_OPTS="--console-address :9001"
+
+# MINIO_SERVER_URL sets the hostname of the local machine for use with the MinIO Server
+# MinIO assumes your network control plane can correctly resolve this hostname to the local machine
+
+# Uncomment the following line and replace the value with the correct hostname for the local machine and port for the MinIO server (9000 by default).
+
+#MINIO_SERVER_URL="http://minio.example.net:9000"
+EOF
+}
+
+generate_tls_certs()
+{
+    if [ -z "${MINIO_HOSTNAME}" ]; then
+        msg info "ONEAPP_MINIO_HOSTNAME is not set or empty. Generating certs with default name minio.example.net"
+        MINIO_HOSTNAME="localhost,minio-*.example.net"
+    fi
+    cd /opt/minio/certs
+    msg info "Generate TLS certificates using certgen"
+    certgen -host "${MINIO_HOSTNAME}"
+
+    if [ $? -eq 0 ]; then
```

Comment:
I think something like 

```shell
if certgen -host "${MINIO_HOSTNAME}"; then
```

should be nicer. :+1:

BTW, is the certgen command idempotent, will it overwrite existing certs when executed for a second time? :thinking: